wsmaster/che-core-api-auth-bitbucket/src/main/java/org/eclipse/che/security/oauth1/NoopOAuthAuthenticator.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.security.oauth1;
+
+import java.net.URL;
+
+/**
+ * Dummy implementation of @{@link OAuthAuthenticator} used in the case if no Bitbucket Server
+ * integration is configured.
+ */
+public class NoopOAuthAuthenticator extends OAuthAuthenticator {
+  protected NoopOAuthAuthenticator() {
+    super(null, null, null, null, null, null, null);
+  }
+
+  @Override
+  String getOAuthProvider() {
+    return "Noop";
+  }
+
+  @Override
+  String getAuthenticateUrl(URL requestUrl, String requestMethod, String signatureMethod)
+      throws OAuthAuthenticationException {
+    throw new RuntimeException("Invalid usage of NoopOAuthAuthenticator");
```

Comment:
`UnsupportedOperationException` or actually `OAuthAuthenticationException`? The message "Invalid usage of NoopOAuthAuthenticator" would suggest to me that there is some kind of valid usage, but there seems to be none. What about "NoopOAuthAuthenticator can't be used for authentication" or maybe even better something like "The fallback noop authenticator cannot be used for authentication. Make sure OAuth is properly configured." or something along those lines to be more specific about why this condition might have arisen?